core/pkg/telemetry/builder.go

```diff
@@ -107,6 +118,38 @@ func BuildConnectOptions(cfg Config) ([]connect.HandlerOption, error) {
 	return options, nil
 }
 
+func buildTransportCredentials(_ context.Context, cfg CollectorConfig) (credentials.TransportCredentials, error) {
+	creds := insecure.NewCredentials()
+	if cfg.KeyPath != "" || cfg.CertPath != "" || cfg.CAPath != "" {
+		capool := x509.NewCertPool()
+		if cfg.CAPath != "" {
+			ca, err := os.ReadFile(cfg.CAPath)
+			if err != nil {
+				return nil, fmt.Errorf("can't read ca file from %s", cfg.CAPath)
+			}
+			if !capool.AppendCertsFromPEM(ca) {
+				return nil, fmt.Errorf("can't add CA '%s' to pool", cfg.CAPath)
+			}
+		}
+
+		tlsConfig := &tls.Config{
+			RootCAs: capool,
+			GetCertificate: func(chi *tls.ClientHelloInfo) (*tls.Certificate, error) {
```

Comment:
for now this just re-reads the files everytime it makes a connection (which isn't very efficient). If this general approach is okay, we can take the same approach as the collector with `reload_interval` (https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md#tls--mtls-configuration)

Reply:
I'm not sure I understand this comment. Wouldn't this code only ever run once on startup?

Reply:
The `GetCertificate` option to `tls.Config` is a func() (in this case a lambda) that will be called every time the TLS connection gets a ClientHello if we don't give it any certificates. Loading the capool above this section will only be done once at startup

https://pkg.go.dev/crypto/tls
```go
// GetCertificate returns a Certificate based on the given
// ClientHelloInfo. It will only be called if the client supplies SNI
// information or if Certificates is empty.
//
// If GetCertificate is nil or returns nil, then the certificate is
// retrieved from NameToCertificate. If NameToCertificate is nil, the
// best element of Certificates will be used.
//
// Once a Certificate is returned it should not be modified.
GetCertificate func(*[ClientHelloInfo](https://pkg.go.dev/crypto/tls#ClientHelloInfo)) (*[Certificate](https://pkg.go.dev/crypto/tls#Certificate), [error](https://pkg.go.dev/builtin#error))
```

Reply:
Ahh, thanks, I missed that was a lambda being passed and not just a one time execution.